### PR TITLE
Add class CartesianPositionStraight

### PR DESCRIPTION
**Pull request by J-Schaefer**
Add class CartesianPositionStraight to ensure a straight movement

Also:

  * Adding a test to open all drawers
  * Add prefix in basic constraints to enable multiple use of the constraints
  * already replaced CartesianPosition by the new class
  * Added some new conversion functions to tf_wrapper

### CODE REVIEW

File: src/giskardpy/constraints.py [hunk 1]
```diff
@@ -910,6 +913,73 @@ def make_constraints(self):
...
+                                    }
+                            }', #required
+            "weight": 1, #optional
+            "max_velocity": 0.3 #optional -- rad/s or m/s depending on joint; can not go higher than urdf limit
+        }'
+        :return:
+        """
+        root_P_goal = w.position_of(self.get_goal_pose())
+        root_P_tip = w.position_of(self.get_fk(self.root, self.tip))
+        root_V_start = w.point3(*tf.transform_stamped_to_pq(tf.lookup_transform(self.root, self.tip))[0])
```
Comment (ichumuh) [inline src/giskardpy/constraints.py:951]:
```
asdf = tf.lookup_transform(self.root, self.tip)
w.point3(asdf.position.x,
asdf.position.y,
asdf.position.z)
```

File: src/giskardpy/constraints.py [hunk 2]
```diff
@@ -910,6 +913,73 @@ def make_constraints(self):
...
+        self.add_minimize_position_constraints(root_P_goal,
+                                               max_velocity,
+                                               max_acceleration,
+                                               self.root,
+                                               self.tip,
+                                               self.goal_constraint,
+                                               weight,
+                                               prefix=u'goal')
+
+        self.add_debug_vector(u'start_point', root_P_goal)
```
Comment (ichumuh) [inline src/giskardpy/constraints.py:966]: no debug

File: src/giskardpy/constraints.py [hunk 3]
```diff
@@ -910,6 +913,73 @@ def make_constraints(self):
...
+        # FIXME: type error. Mixed calc with Vector3 and SX (expression)
```
Comment (ichumuh) [inline src/giskardpy/constraints.py:967]: fixme?

File: src/giskardpy/constraints.py [hunk 4]
```diff
@@ -1198,7 +1268,7 @@ def __init__(self, god_map, root_link, tip_link, goal, translation_max_velocity=
                  weight=WEIGHT_ABOVE_CA, goal_constraint=True):
         super(CartesianPose, self).__init__(god_map)
         self.constraints = []
-        self.constraints.append(CartesianPosition(god_map=god_map,
+        self.constraints.append(CartesianPositionStraight(god_map=god_map,
```
Comment (ichumuh) [inline src/giskardpy/constraints.py:1271]: create a `CartesianPoseStraight` for now

File: src/giskardpy/python_interface.py [hunk 5]
```diff
@@ -61,14 +61,14 @@ def set_translation_goal(self, root, tip, pose_stamped, weight=None, max_velocit
...
-            constraint.type = CartesianConstraint.TRANSLATION_3D
+            constraint.type = CartesianConstraint.STRAIGHT_TRANSLATION_3D
             constraint.root_link = str(root)
             constraint.tip_link = str(tip)
             constraint.goal = pose_stamped
             self.cmd_seq[-1].cartesian_constraints.append(constraint)
         else:
             constraint = Constraint()
-            constraint.type = u'CartesianPosition'
+            constraint.type = u'CartesianPositionStraight'
```
Comment (ichumuh) [inline src/giskardpy/python_interface.py:71]: add functions for straight option

File: src/giskardpy/qp_problem_builder.py [hunk 6]
```diff
@@ -298,7 +298,7 @@ def get_cmd(self, substitutions, nWSR=None):
         if xdot_full is None:
             return None
         # TODO enable debug print in an elegant way, preferably without slowing anything down
-        # self.debug_print(np_H, A, lb, ub, lbA, ubA, g, xdot_full)
+        self.debug_print(np_H, A, lb, ub, lbA, ubA, g, xdot_full)
```
Comment (ichumuh) [inline src/giskardpy/qp_problem_builder.py:301]: no debug

File: src/giskardpy/tfwrapper.py [hunk 7]
```diff
@@ -350,6 +350,88 @@ def np_vector(x, y, z):
 def np_point(x, y, z):
     return np.array([x, y, z, 1])
 
+### Code copied from user jarvisschultz from ROS answers
+#   https://answers.ros.org/question/332407/transformstamped-to-transformation-matrix-python/
+def pose_to_pq(msg):
```
Comment (ichumuh) [inline src/giskardpy/tfwrapper.py:355]: rename to pose_to_numpy

File: src/giskardpy/tfwrapper.py [hunk 8]
```diff
@@ -350,6 +350,88 @@ def np_vector(x, y, z):
...
+    """Convert a C{geometry_msgs/Pose} into position/quaternion np arrays
+
+    @param msg: ROS message to be converted
+    @return:
+      - p: position as a np.array
+      - q: quaternion as a numpy array (order = [x,y,z,w])
+    """
```
Comment (ichumuh) [inline src/giskardpy/tfwrapper.py:362]: use my comment style

File: src/giskardpy/tfwrapper.py [hunk 9]
```diff
@@ -350,6 +350,88 @@ def np_vector(x, y, z):
...
+    return transform_to_pq(msg.transform)
+
+
+def msg_to_se3(msg):
+    """Conversion from geometric ROS messages into SE(3)
```
Comment (ichumuh) [inline src/giskardpy/tfwrapper.py:406]: rename to `msg_to_homogeneous_matrix`

File: test/test_integration_pr2.py [hunk 10]
```diff
@@ -1170,7 +1172,97 @@ def test_open_drawer(self, kitchen_setup):
...
+                       u'kitchen_island_left_lower_drawer_handle',
+                       u'kitchen_island_middle_upper_drawer_handle',
+                       u'kitchen_island_middle_lower_drawer_handle',
+                       u'kitchen_island_right_upper_drawer_handle',
+                       u'kitchen_island_right_lower_drawer_handle',
+                       u'oven_area_area_left_drawer_handle',
+                       u'oven_area_area_right_drawer_handle']
+
+        handle_frame_id = [u'iai_kitchen/' + item for item in handle_name]
+        joint_name = [str(item).decode("utf-8").replace(u'handle', u'main_joint').encode("utf-8") for item in handle_name]
```
Comment (ichumuh) [inline test/test_integration_pr2.py:1198]: too complicated.

File: test/test_integration_pr2.py [hunk 11]
```diff
@@ -1170,7 +1172,97 @@ def test_open_drawer(self, kitchen_setup):
...
+
+        for i_handle_id, i_handle_name, i_joint_name in zip(handle_frame_id, handle_name, joint_name):
+            rospy.loginfo('=== Opening drawer: ' + str(i_handle_name).decode('utf-8') + ' ===')
```
Comment (ichumuh) [inline test/test_integration_pr2.py:1201]: `logging.loginfo`
use normal unicode

File: src/giskardpy/constraints.py (hunk 1 above)
Comment (ichumuh) [reply src/giskardpy/constraints.py:951]: use `get_input_PoseStamped` and write tf pose to godmap in init

**Review by ichumuh — CHANGES_REQUESTED** [review]

**Comment by J-Schaefer** [comment]
I changed everything. Free for testing now!

**Comment by J-Schaefer** [comment]
Is this still current? There are a few conflicts but only regarding names. Did you have issues last time, @ichumuh ??
The tests work on my system although ```test_CartesianPoseStraight``` freezes and only works when added in the function ```set_translation_goal``` in ```python_interface```.

**Comment by ichumuh** [comment]
I didn't merge it because of the error i posted here: https://github.com/SemRoCo/giskard_msgs/pull/14
Probably didn't post the error here by accident.
Not sure if that is fixed now. If you merge with devel, I'll try again.

**Comment by J-Schaefer** [comment]
I remember, you mentioned before. I'll check.

**Pull request merged**